### PR TITLE
[Celo integration] Fix amount calculation in send max when fees exceed spendable amount

### PR DESCRIPTION
**Pull request by pawelnguyen**
## Context (issues, jira)
This fixes amount calculation in send max when fees exceed spendable amount. In this case we just set amount to 0.

### Before:
CleanShot 2022-04-19 at 12 07 31

### After:
CleanShot 2022-04-19 at 12 10 03

## Expectations

- [ ] **Test coverage: The changes of this PR are covered by test.** Unit test were added with mocks when depending on a backend/device.
- [ ] **No impact: The changes of this PR have ZERO impact on the userland.** Meaning, we can use these changes without modifying LLD/LLM at all. It will be a "noop" and the maintainers will be able to bump it without changing anything.

### CODE REVIEW

**Comment by pawelnguyen** [comment]
@haammar-ledger @henrily-ledger 🙇

**Review by henri-ly — APPROVED** [review]

**Comment by henrily-ledger** [comment]
@pawelnguyen is it possible to sign the commit otherwise we can't merge it :(

**Comment by pawelnguyen** [comment]
@henrily-ledger sure, commit signed :)

**Comment by henrily-ledger** [comment]
Ok we are now doing QA on our side to be sure everything is alright ! Once it done we will merged it

**Pull request merged**
